services/web/server/src/simcore_service_webserver/projects/_crud_api_read.py

```diff
@@ -143,25 +143,54 @@ async def list_projects(  # pylint: disable=too-many-arguments
 
 
 async def list_projects_full_search(
-    app,
+    request,
     *,
     user_id: UserID,
     product_name: str,
     offset: NonNegativeInt,
     limit: int,
     text: str | None,
+    order_by: OrderBy,
+    tag_ids_list: list[int],
 ) -> tuple[list[ProjectDict], int]:
-    db = ProjectDBAPI.get_from_app_context(app)
+    db = ProjectDBAPI.get_from_app_context(request.app)
+
+    user_available_services: list[dict] = await get_services_for_user_in_product(
+        request.app, user_id, product_name, only_key_versions=True
+    )
 
-    total_number_projects, db_projects = await db.list_projects_full_search(
+    (
+        db_projects,
+        db_project_types,
+        total_number_projects,
+    ) = await db.list_projects_full_search(
         user_id=user_id,
         product_name=product_name,
+        filter_by_services=user_available_services,
         text=text,
         offset=offset,
         limit=limit,
+        order_by=order_by,
+        tag_ids_list=tag_ids_list,
     )
 
-    return db_projects, total_number_projects
+    projects: list[ProjectDict] = await logged_gather(
```

Comment:
Why are we still using `logged_gather`? Is there any reason why you want to log every call?
@sanderegg did a much reliable and safe version of this in `limited_gather`.  IMO we should have deprecated 
Check the details (see e.g. tests) and let's start using this new version in all new code. It is not only

Reply:
I will budge in on this one and say it's not clear to me the use case. I do not perceive `limited_gather` as an alternative to `logged_gather`, it has a different behaviour by default which is pretty bad. It limits parallelism to 1. I understand the benefits but this breaks the deal for me.

If you want to mark `logged_gather` as deprecated, then remove the default value for `limit` in  `limited_gather` and let the user specify it each time. I think that's a better approach, since it will avoid hidden behaviours, like running tasks sequentially, which is not the intention when using that function.

Reply:
Unfortunately it doesn't work as one to one replacement so I will not change it now. I will investigate/ask Sylvain when he is back.